### PR TITLE
chore: normalize status.observedState.etag in golden tests

### PR DESCRIPTION
**Pull request by justinsb**
Also simplify etag computation, as we don't rely on the value itself
any more.

### CODE REVIEW

**Comment by justinsb** [comment]
(I did this because it blocked https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/2052 )

File: mockgcp/common/fields/etag.go [hunk 1]
```diff
@@ -19,53 +19,17 @@ import (
...
-		panic(fmt.Sprintf("converting to prototext: %v", err))
+		panic(fmt.Sprintf("converting to proto: %v", err))
 	}
-	h := sha256.Sum256([]byte(m))
-	str := base64.StdEncoding.EncodeToString(h[:])
-	strong := fmt.Sprintf(`"%s"`, str) // ETag must be quoted.
-	return "W/" + strong
-}
-
-func ComputeEtagBytes(obj proto.Message) []byte {
```
Comment (yuwenma) [inline mockgcp/common/fields/etag.go:61]: I checked the current in-use `ComputeEtag`. Some expects a []byte form and some needs a pointer. So I'm trying to make this a generic lib for other  contributors to use out of the box, so they don't need to handle the dereference pointer badly (like what I did previously).

**Comment by yuwenma** [comment]
/lgtm
/approve



> (I did this because it blocked #2052 )

Ah! Fair

**Pull request merged**